### PR TITLE
Sodium + Optifine support

### PR DESCRIPTION
**Pull request by Rubydesic**

### CODE REVIEW

**Comment by Rubydesic** [comment]
This also supports Rubidium/Oculus, forge forks of Sodium/Iris

File: common/src/main/java/org/valkyrienskies/mod/mixin/client/world/MixinClientChunkCache.java [hunk 1]
```diff
@@ -79,8 +85,10 @@ private void preLoadChunkFromPacket(final int x, final int z, final ChunkBiomeCo
     @Inject(method = "drop", at = @At("HEAD"), cancellable = true)
     public void preUnload(final int chunkX, final int chunkZ, final CallbackInfo ci) {
         shipChunks.remove(ChunkPos.asLong(chunkX, chunkZ));
-        ((IVSViewAreaMethods) ((LevelRendererAccessor) ((ClientLevelAccessor) level).getLevelRenderer()).getViewArea())
-            .unloadChunk(chunkX, chunkZ);
+        if (ValkyrienCommonMixinConfigPlugin.getVSRenderer() != VSRenderer.SODIUM) {
```
Comment (StewStrong) [inline common/src/main/java/org/valkyrienskies/mod/mixin/client/world/MixinClientChunkCache.java:88]: Do we need to call something similar for Sodium as well?

File: common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/sodium/MixinChunkRenderManager.java [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+    private void addChunkToSpecificRenderList(final ChunkRenderContainer<T> render,
+        final ChunkRenderList<T>[] renderList, final Vector3dc camInShip) {
+        // todo compute visible faces correctly
+        final int visibleFaces = this.computeVisibleFaces(render, camInShip) & render.getFacesWithData();
+        if (visibleFaces != 0) {
+            boolean added = false;
+            final T[] states = render.getGraphicsStates();
+
+            outer:
```
Comment (StewStrong) [inline common/src/main/java/org/valkyrienskies/mod/mixin/mod_compat/sodium/MixinChunkRenderManager.java:259]: Do we need this label?

File: fabric/src/main/resources/fabric.mod.json [hunk 3]
```diff
@@ -33,7 +33,5 @@
     "minecraft": ">=1.16.5"
   },
   "breaks": {
-    "sodium": "*",
```
Comment (StewStrong) [inline fabric/src/main/resources/fabric.mod.json:36]: Nice! :)

File: common/src/main/java/org/valkyrienskies/mod/mixin/client/world/MixinClientChunkCache.java (hunk 1 above)
Comment (Rubydesic) [reply common/src/main/java/org/valkyrienskies/mod/mixin/client/world/MixinClientChunkCache.java:88]: fixed

**Review by StewStrong — APPROVED** [review]
Very cool!

**Pull request merged**